imod/mf6/simulation.py

```diff
@@ -916,21 +917,21 @@ def split(self, submodel_labels: xr.DataArray) -> Modflow6Simulation:
             new_simulation[package_name] = package
 
         for model_name, model in original_models.items():
+            solution_name = self.get_solution_name(model_name)
+            new_simulation[solution_name].remove_model_from_solution(model_name)
             for submodel_partition_info in partition_info:
                 new_model_name = f"{model_name}_{submodel_partition_info.id}"
                 new_simulation[new_model_name] = slice_model(
                     submodel_partition_info, model
                 )
+                new_simulation[solution_name].add_model_to_solution(new_model_name)
 
         exchanges = []
         for model_name, model in original_models.items():
-            exchanges += exchange_creator.create_exchanges(
-                model_name, model.domain.layer
-            )
-
-        new_simulation["solver"]["modelnames"] = xr.DataArray(
-            list(get_models(new_simulation).keys())
-        )
+            if isinstance(model, GroundwaterFlowModel):
```

Comment:
An alternative to an if statement if passing the entire model to the method and then use typed dispatch to call the correct method
```
for model_name, model in original_models.items():
    exchanges += exchange_creator.create_exchanges(
        model_name, model
    )
```

In the ExchangeCreator:
```
@typedispatch
def create_exchanges(
        self, model_name: str, model: GroundwaterFlowModel
    ) -> List[GWFGWF]:
....

@typedispatch
def create_exchanges(
        self, model_name: str, model: GroundwaterTransportModel
    ) -> List[GWFGWF]:
    
    return []
```